### ISSUE TITLE
B&O Beoplay H4 headset - Not battery repported

### ISSUE THREAD

**Issue by rafaelbartolome** [jariz/Akku#31]
First of all, great project, thanks for sharing it.

Sadly I can't make it work with my headphones. Here is the tech info just in case:

MacOS : Mojave 10.14.6
Akku : Release 0.1.0-beta.8
Headset : Bang&Olufsen Beoplay H4

```
$ pgrep -fli akku
95946 /Applications/Akku.app/Contents/MacOS/Akku
95947 /Library/PrivilegedHelperTools/io.jari.AkkuHelper
```

```
$ system_profiler SPBluetoothDataType
2019-08-08 12:55:30.893 system_profiler[97341:7022214] SystemInfo-AccessoryFW from dict - Beoplay H4 = 0x0002
...
          Beoplay H4:
              Address: 00-09-A7-13-C2-E2
              Major Type: Audio
              Minor Type: Headphones
              Services: Headset, Hands-Free unit
              Paired: Yes
              Configured: Yes
              Connected: Yes
              Manufacturer: Cambridge Silicon Radio (0x8, 0x2C81)
              Firmware Version: 0x0002
              Vendor ID: 0x0103
              Product ID: 0x1008
              Class of Device: 0x04 0x06 0x240418
              RSSI: -53
              Role: Master
              Connection Mode: Active Mode
              Interval: 0 ms
              EDR Supported: Yes
              eSCO Supported: Yes
              SSP Supported: Yes
```

I can't find any of the log files, or crash logs in the system, but I think the info needed is in the [bluetooth dump](https://wetransfer.com/downloads/cd1dc63fc53725479efb0e718e7b745620190808110308/28fd87)

As far as I can understand the battery info is there:
```
AT+IPHONEACCEV=2,1,6,2,0
Frame 49123: 38 bytes on wire (304 bits), 38 bytes captured (304 bits)
Bluetooth
PacketLogger Recv ACL Data
Bluetooth HCI ACL Packet
    .... 0000 0000 1101 = Connection Handle: 0x00d
    ..10 .... .... .... = PB Flag: First Automatically Flushable Packet (2)
    00.. .... .... .... = BC Flag: Point-To-Point (0)
    Data Total Length: 33
    Data
    [Connect in frame: 48562]
    [Source BD_ADDR: BangOluf_13:c2:e2 (00:09:a7:13:c2:e2)]
    [Source Device Name: Beoplay H4]
    [Source Role: Slave (2)]
    [Destination BD_ADDR: 00:00:00_00:00:00 (00:00:00:00:00:00)]
    [Destination Device Name: ]
    [Destination Role: Master (1)]
    [Last Role Change in Frame: 48534]
    [Current Mode: Active Mode (0)]
    [Last Mode Change in Frame: 48562]
Bluetooth L2CAP Protocol
    Length: 29
    CID: Dynamically Allocated Channel (0x0042)
    [Connect in frame: 48867]
    [PSM: RFCOMM (0x0003)]
Bluetooth RFCOMM Protocol
    Address: E/A flag: 1, C/R flag: 0, Direction: 0, Channel: 10
    Control: Frame type: Unnumbered Information with Header check (UIH) (0xef), P/F flag: 0
    Payload length: 25
    Frame Check Sequence: 0xeb
Bluetooth HFP Profile
    [Role: HS - Headset (2)]
    AT Stream: AT+IPHONEACCEV=2,1,6,2,0\r
    Command 0: +IPHONEACCEV
        Command Line Prefix: AT
        Command: +IPHONEACCEV (Apple Bluetooth Headset Battery Level Indication)
        Type: Action Command (0x003d)
        Parameters
            Count: 2
            Key: Battery Level (1)
            Value: 6
            Key: Dock State (2)
            Value: 0
```

Thanks.

**Comment by jariz** [comment]
Hiya, thanks for the dump and your interest in the project. 
It appears somewhere between 10.14.2 and 10.14.6 some changes were made to the system akku uses to read it's packets from that caused it to break.
I'm working on it now, will keep you updated.

**Comment by jariz** [comment]
(see also: #30)

**Comment by rafaelbartolome** [comment]
Looks like [10cf410](https://github.com/jariz/Akku/commit/10cf410d022b139dd56220d9e251334421c9192d) solves de issue.

![Screenshot 2019-08-09 at 09 32 37](https://user-images.githubusercontent.com/5570735/62762123-a64ea000-ba88-11e9-9231-821517b4b07c.png)

Waiting for the next release!!!

Thanks.

**Comment by jariz** [comment]
Awesome!

**Comment by jariz** [comment]
released - 0.1.0-beta.9
